maven-plugin/src/main/java/org/openmrs/maven/plugins/AddDependency.java

```diff
@@ -0,0 +1,134 @@
+     */
+    @Parameter(property = "type")
+    private String type;
+
+    /**
+     * Maven group id of the dependency
+     */
+    @Parameter(property = "groupId")
+    private String groupId;
+
+    /**
+     * Maven artifact id of the dependency
+     */
+    @Parameter(property = "artifactId")
+    private String artifactId;
+
+    /**
+     * Version of the dependency
+     */
+    @Parameter(property = "version")
+    private String version;
+
+    /**
+     * Name of the frontend module
+     */
+    @Parameter(property = "moduleName")
+    private String moduleName;
+
+
+    private final String DEPENDENCY_TYPE_PROMPT = "Enter the type of dependency you need to add";
+
+
+    @Override
+    public void executeTask() throws MojoExecutionException, MojoFailureException  {
+        if (distro == null) {
+            File userDir = new File(System.getProperty("user.dir"));
+            File distroFile = new File(userDir, DistroProperties.DISTRO_FILE_NAME);
+            if (distroFile.exists()) {
+                distro = distroFile.getAbsolutePath();
+            }
+        }
+
+        if(StringUtils.isBlank(type)) {
+            List<String> dependencyTypes = new ArrayList<>(Arrays.asList("OMOD", "SPA", "OWA", "WAR"));
+            type = wizard.promptForMissingValueWithOptions(DEPENDENCY_TYPE_PROMPT, null, null, dependencyTypes);
+        }
+        Properties properties = new Properties();
+
+        if(StringUtils.isNotBlank(distro)) {
+            properties = distroHelper.resolveDistroPropertiesForStringSpecifier(distro, versionsHelper).getProperties();
+        }
+
+        switch (type) {
+            case "OMOD":
+                groupId = wizard.promptForValueIfMissingWithDefault(null, groupId, "groupId", Artifact.GROUP_MODULE);
+                artifactId = wizard.promptForValueIfMissing(artifactId, "artifactId");
+                if (StringUtils.isBlank(version)) {
+                    Artifact artifact = new Artifact(artifactId, "1.0", groupId);
+                    List<String> versions = versionsHelper.getSuggestedVersions(artifact, 5);
+                    version = wizard.promptForMissingValueWithOptions("Enter the version", null, null, versions,
+                            "Please specify the module version", null);
+                }
+                properties.setProperty("omod." + artifactId, version);
+                break;
+            case "SPA":
+                moduleName = wizard.promptForValueIfMissing(moduleName, "frontend module name");
+                if (StringUtils.isBlank(version)) {
+                    List<String> versions = new NPMHelper().getPackageVersions(moduleName, 6);
+                    version = wizard.promptForMissingValueWithOptions("Enter the module version", null, null, versions,
+                            "Please specify the SPA version", null);
+                }
+                properties.setProperty("spa.frontendModules.@openmrs/" + moduleName, version);
```

Comment:
The `@openmrs/` part _is_ a part of the module name (and we need to allow for implementations that can't publish to our NPM org) 

```suggestion
                properties.setProperty("spa.frontendModules." + moduleName, version);
```